generators/chipyard/src/main/scala/SpikeTile.scala

```diff
@@ -0,0 +1,392 @@
+import freechips.rocketchip.interrupts._
+import freechips.rocketchip.util._
+import freechips.rocketchip.tile._
+import freechips.rocketchip.prci.ClockSinkParameters
+
+case class SpikeCoreParams(
+  val maxInsnsPerCycle: Int = 10000
+) extends CoreParams {
+  val useVM = true
+  val useHypervisor = false
+  val useSupervisor = true
+  val useUser = true
+  val useDebug = true
+  val useAtomics = true
+  val useAtomicsOnlyForIO = false
+  val useCompressed = true
+  override val useVector = true
+  val useSCIE = false
+  val useRVE = false
+  val mulDiv = Some(MulDivParams())
+  val fpu = Some(FPUParams())
+  val nLocalInterrupts = 0
+  val useNMI = false
+  val nPTECacheEntries = 0
+  val nPMPs = 16
+  val pmpGranularity = 4
+  val nBreakpoints = 0
+  val useBPWatch = false
+  val mcontextWidth = 0
+  val scontextWidth = 0
+  val nPerfCounters = 0
+  val haveBasicCounters = true
+  val haveFSDirty = true
+  val misaWritable = true
+  val haveCFlush = false
+  val nL2TLBEntries = 0
+  val nL2TLBWays = 0
+  val mtvecInit = None
+  val mtvecWritable = true
+  val instBits = 16
+  val lrscCycles = 1
+  val decodeWidth = 1
+  val fetchWidth = 1
+  val retireWidth = 1
+  val bootFreqHz = BigInt(1000000000)
+  val rasEntries = 0
+  val btbEntries = 0
+  val bhtEntries = 0
+
+  override def vLen = 128
+  override def vMemDataBits = 128
+}
+
+case class SpikeTileAttachParams(
+  tileParams: SpikeTileParams
+) extends CanAttachTile {
+  type TileType = SpikeTile
+  val lookup = PriorityMuxHartIdFromSeq(Seq(tileParams))
+  val crossingParams = RocketCrossingParams()
+}
+
+case class SpikeTileParams(
+  hartId: Int = 0,
+  val core: SpikeCoreParams = SpikeCoreParams(),
+  icacheParams: ICacheParams = ICacheParams(nWays = 32),
+  dcacheParams: DCacheParams = DCacheParams(nWays = 32)
+) extends InstantiableTileParams[SpikeTile]
+{
+  val name = Some("spike_tile")
+  val beuAddr = None
+  val blockerCtrlAddr = None
+  val btb = None
+  val boundaryBuffers = false
+  val dcache = Some(dcacheParams)
+  val icache = Some(icacheParams)
+  val clockSinkParams = ClockSinkParameters()
+  def instantiate(crossing: TileCrossingParamsLike, lookup: LookupByHartIdImpl)(implicit p: Parameters): SpikeTile = {
+    new SpikeTile(this, crossing, lookup)
+  }
+}
+
+class SpikeTile(
+  val spikeTileParams: SpikeTileParams,
+  crossing: ClockCrossingType,
+  lookup: LookupByHartIdImpl,
+  q: Parameters) extends BaseTile(spikeTileParams, crossing, lookup, q)
+    with SinksExternalInterrupts
+    with SourcesExternalNotifications
+{
+  // Private constructor ensures altered LazyModule.p is used implicitly
+  def this(params: SpikeTileParams, crossing: TileCrossingParamsLike, lookup: LookupByHartIdImpl)(implicit p: Parameters) =
+    this(params, crossing.crossingType, lookup, p)
+
+  // Required TileLink nodes
+  val intOutwardNode = IntIdentityNode()
+  val masterNode = visibilityNode
+  val slaveNode = TLIdentityNode()
+
+  override def isaDTS = "rv64gcv_Zfh"
```

Comment:
Is this true with the current CY toolchain?

Reply:
This isn't passed to the toolchain.

Reply:
I know it's not passed but right now we can't compile code outside `gc` right? I'm not sure it would make sense to restrict this to `rv64gc` for now.

Reply:
rv64gcv_Zfh is a superset of rv64gc. The cpu is always allowed to report that it has more capability than the software needs.